### PR TITLE
Worker: Github + Spotify Reactions

### PR DESCRIPTION
**Pull request by Arthi-chaud**

### CODE REVIEW

**Review by 0Nom4D — APPROVED** [review]

File: worker/src/services/spotify.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+	private _spotify;
+
+	constructor(_: Pipeline) {
+		super();
+		// TODO load credentials
+		this._spotify = new SpotifyWebApi();
+	}
+
+	private async searchTrack(artistName: string, trackName: string)  {
```
Comment (AnonymusRaccoon) [inline worker/src/services/spotify.ts:16]: privates should start with a _

File: worker/src/services/spotify.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		return this._spotify.searchTracks(`artist=${artistName}&track=${trackName}`).then(
```
Comment (AnonymusRaccoon) [inline worker/src/services/spotify.ts:17]: You should not return a Promise with a then in an async's body. Simply use await.

File: worker/src/services/spotify.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+				if (searchResult.playlists.total == 0)
+					throw new Error(`Spotify API: '${playlistName}': no such playlist`);
+				return searchResult.playlists.items[0];
+			}
+		);
+	}
+
+	@reaction(ReactionType.playTrack, ['artist', 'track'])
+	playTrack(params: any) {
+		this.searchTrack(params['artist'], params['track']).then(
```
Comment (AnonymusRaccoon) [inline worker/src/services/spotify.ts:38]: Should't we take an ID for that instead of searching? It would prevent false positives.

**Review by AnonymusRaccoon — CHANGES_REQUESTED** [review]
You should use await instead of then and use arrow functions instead of the function keyword.

File: worker/src/services/spotify.ts (hunk 3 above)
Comment (Arthi-chaud) [reply worker/src/services/spotify.ts]: The tracks' ID are now shown on user's interfaces (mobile app), therefore it is not easy for a set to get it. The easiest way for a user to use the reaction is by using the track's name/artist

**Review by AnonymusRaccoon — APPROVED** [review]

**Review by GitBluub — APPROVED** [review]

**Pull request merged**